src/ast/passes/semantic_analyser.cpp

```diff
@@ -817,6 +818,31 @@ void SemanticAnalyser::visit(Call &call)
 
     call.type = CreateInet(buffer_size);
   }
+  else if (call.func == "pton")
+  {
+    if (!check_nargs(call, 1))
+      return;
+    std::string addr = bpftrace_.get_string_literal(call.vargs->at(0));
```

Comment:
This might be an empty string. Is the current error (one that `get_string_literal()` may print) good enough or can we give a nicer one?

Reply:
Thanks @danobi . If `addr` is an empty string, this would be catched by the `else` branch at line 840, which prints `ERROR: pton() expects an string argument of an IPv4/IPv6 address, got`. Should I add another check above?

Reply:
Ah, I see. No, this is fine